src/sentry/sentry_metrics/indexer/base.py

```diff
@@ -94,11 +117,51 @@ def as_strings(self) -> Sequence[str]:
         return keys
 
 
+class UseCaseKeyCollection:
+    def __init__(self, mapping: Mapping[str, Union[Mapping[int, Set[str]], KeyCollection]]):
```

Comment:
Why do you need to support both map and keycollection here?

Reply:
keycollection is there in situations where we already have `keycollection` instead of `Map[int, Set[str]]`, for instance [here](https://github.com/getsentry/sentry/blob/d0d153157f1bc521396adc6fa924f80af686b544/src/sentry/sentry_metrics/indexer/base.py#L310).
`unmapped_result` is a `keycollection`, if we don't support `keycollection` we will have to disassemble it into a `Map[int, Set[str]]` before passing it into this constructor just for it to reassemble it into a `keycollection` again.